src/rust/engine/src/context.rs

```diff
@@ -127,6 +129,12 @@ impl Core {
       })
       .unwrap_or_else(|e| panic!("Could not initialize Store: {:?}", e));
 
+    let process_execution_metadata = ExecuteProcessRequestMetadata {
+      instance_name: remote_instance_name.clone(),
+      cache_key_gen_version: remote_execution_process_cache_namespace.clone(),
+      platform_properties: remote_execution_extra_platform_properties.clone(),
+    };
+
     let mut command_runner: Box<dyn process_execution::CommandRunner> =
```

Comment:
Why not have an if/else here, instead of creating the mutable variable?

Reply:
I think my instinct here is because this is more clear in the face of multiple wrappers:
```rust
let mut val = foo;
if wrap {
  val = do_wrap(val);
}
if wrap_in_other {
  val = do_other_wrap(val);
}
```

but I don't have a strong preference either way, if you do.